### ISSUE TITLE
text_layer_get_max_used_size() sometimes returning wrong size

### ISSUE THREAD

**Issue by schneidr** [pebble/pebblekit#17]
I'm adding and removing a text layer to show notifications. To adapt the size of the notification to the text in it I use `text_layer_get_max_used_size()` to get the size of the text and calculate the necessary size of the notification. The problem: sometimes the returned height is wrong. It seems to follow this pattern:
1. correct height
2. wrong height
3. correct height
4. wrong height
   always wrong height from now on.

This happens only as long as a I assign the same text. When I assign a different text it starts over.

Sample code:

``` c
#include "pebble_os.h"
#include "pebble_app.h"
#include "pebble_fonts.h"


#define MY_UUID { 0x4C, 0xD9, 0x45, 0x25, 0xF5, 0x20, 0x40, 0x52, 0x80, 0x84, 0x87, 0x10, 0x46, 0x57, 0x9D, 0x66 }
PBL_APP_INFO(MY_UUID,
             "Template App", "Your Company",
             1, 0, /* App version */
             DEFAULT_MENU_ICON,
             APP_INFO_STANDARD_APP);

#define FRAME_NOTIFICATION (GRect(5, 49, 144-10, 69))
#define TXT_DEMO "Sample text"

Window window;
TextLayer layer_notification;
bool visible;

void show_notification(char* text) {
  GContext *ctx = app_get_current_graphics_context();
  text_layer_set_text(&layer_notification, text);
  GSize size = text_layer_get_max_used_size(ctx, &layer_notification);
  size.w = 134;
  size.h += 10;
  text_layer_set_size(&layer_notification, size);
  layer_add_child(&window.layer, (Layer *)&layer_notification);
}

void hide_notification() {
  layer_remove_from_parent((Layer *)&layer_notification);
}

void select_single_click_handler(ClickRecognizerRef recognizer, Window *window) {
  if (visible) {
    hide_notification();
    visible = false;
  }
  else {
    show_notification(TXT_DEMO);
    visible = true;
  }
}

void config_provider(ClickConfig **config, Window *window) {
  config[BUTTON_ID_SELECT]->click.handler = (ClickHandler) select_single_click_handler;
  config[BUTTON_ID_SELECT]->click.repeat_interval_ms = 1000;
}

void handle_init(AppContextRef ctx) {
  (void)ctx;

  window_init(&window, "Window Name");
  window_stack_push(&window, true /* Animated */);

  window_set_click_config_provider(&window, (ClickConfigProvider) config_provider);

  text_layer_init(&layer_notification, FRAME_NOTIFICATION);
  text_layer_set_background_color(&layer_notification, GColorBlack);
  text_layer_set_text_color(&layer_notification, GColorWhite);
  text_layer_set_font(&layer_notification, fonts_get_system_font(FONT_KEY_GOTHIC_28));
  text_layer_set_text_alignment(&layer_notification, GTextAlignmentCenter);
  text_layer_set_overflow_mode(&layer_notification, GTextOverflowModeWordWrap);
}

void pbl_main(void *params) {
  PebbleAppHandlers handlers = {
    .init_handler = &handle_init
  };
  app_event_loop(params, &handlers);
}
```

Press "Select" on the pebble repeatedly to show and hide the notification.

**Comment by iosdeveloper** [comment]
I experienced issues with this API as well. In my case it was always missing one line of text and I had to add 20 pixels to the height of the returned size. Setting the size of my text layer directly to the returned size would cut the last line of text.

**Comment by iosdeveloper** [comment]
Appears to be fixed in v1.12. Please check again and eventually close this issue.

**Comment by schneidr** [comment]
I can confirm that it seems to be fixed in v1.12.
